client/src/components/Game/Timer.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import { withFirebase } from '../Firebase';
+import { withRouter } from 'react-router';
+import { compose } from 'recompose';
+
+class Timer extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      minutes: 99,
+      seconds: 99,
+      initialSet: false
+    }
+  }
+
+  async componentDidMount() {
+    const { id } = this.props.match.params;
+    const game = this.props.firebase.game(id);
+
+    game.get()
+    .then(docSnapshot => {
+      if(docSnapshot.exists) {
+        game.onSnapshot((snapshot) => {
+          if(snapshot.data().gameStartTime !== null) {
+            this.setState({
+              gameSize: snapshot.data().players.length,
+              startTime: snapshot.data().gameStartTime.seconds,
+              timePerTurnInSeconds: parseInt(snapshot.data().timeLimit, 10) * 60,
+            })
+            this.setState({
+              timeTurnWillEnd: ((this.state.timePerTurnInSeconds * this.state.gameSize) + this.state.startTime),
+              currentTime: Math.floor(new Date().getTime() / 1000),
+            })
+            this.setState({
+              minutes: Math.floor((this.state.timeTurnWillEnd - this.state.currentTime)/60),
+              seconds: (this.state.timeTurnWillEnd - this.state.currentTime)%60
+            })
+          }
+        })
+      }
+    })
+
+    this.myInterval = setInterval(async () => {
+      const { seconds, minutes } = this.state
+      const gameRef = await this.props.firebase.game(id).get();
+
+      if (seconds > 0) {
+        this.setState(({ seconds }) => ({
+          seconds: seconds - 1
+        }))
```

Comment:
Don't think you need this function here. Looks like you can just set the seconds state directly.